src/pages/BlendPoolPage.tsx

```diff
@@ -111,6 +125,47 @@ export default function BlendPoolPage() {
     };
   }, [poolData]);
 
+  useEffect(() => {
```

Comment:
Decided to add this here rather than trying to move the logic outside of the components since it should be cached and thus require fewer requests.